### DISCUSSION TITLE
Disable steppers after printing and is mesh automatically activated ?

### DISCUSSION THREAD

**Discussion by fish7221** [mriscoc/Ender3V2S1#906] [Q&A] [answered]
Hello. This is my first post so i hope I post it properly.
First of all thank you SO MUCH Mriscoc for this great and useful firmware.
I just flashed the 03/23 firmware on my Ender 3 V2 and I noticed 2 things changed since 2022 firmware I've used before :
1. In the end of the print the steppers are enabled (I didn't find an option to disable it).
2. In the previous firmware, After mesh leveling, Mesh was only activated on the first printing and then needed to be activated manually, And I see that option is gone. Does it means that now mesh will be activated automatically ?

Thank you

**Comment by mriscoc** [comment, marked as answer]
> 1. In the end of the print the steppers are enabled (I didn't find an option to disable it).

Use the end G-code script to disable the motors and heaters if it is required, now you have the choice.

```gcode
M106 S0 ;Turn-off fan
M104 S0 ;Turn-off hotend
M140 S0 ;Turn-off bed
M84 ;Disable steppers
```
> 2\. In the previous firmware, After mesh leveling, Mesh was only activated on the first printing and then needed to be activated manually, And I see that option is gone. Does it means that now mesh will be activated automatically ?

Use `M420 S1` or (`G29 A` for UBL only) after the G28 line in your start G-code script, a valid mesh must be loaded in to memory first.

**Comment by fish7221** [comment]
Thank you so much for the quick answers.
